qualtran/bloqs/comparison_gates.ipynb

```diff
@@ -0,0 +1,957 @@
+    "    qubit_order += target_qubits\n",
+    "    n_qubits = len(data_qubits)\n",
+    "    if not quantum_classical:\n",
+    "        n_qubits >>= 1\n",
+    "    for v in range(1 << len(data_qubits)):\n",
+    "        bits = [(v >> i) & 1 for i in range(len(data_qubits) - 1, -1, -1)]\n",
+    "        bits += (len(qubit_order) - len(data_qubits)) * [0]\n",
+    "        result = sim.simulate(c, qubit_order=qubit_order, initial_state=bits)\n",
+    "        s = f'final state vector of {v} compared to {qn}'\n",
+    "        if not quantum_classical:\n",
+    "            msk = (1 << n_qubits) - 1\n",
+    "            s = f'final state vector of {v >> n_qubits} compared to {v & msk}'\n",
+    "        print(\n",
+    "            s,\n",
+    "            format_dirac(result.dirac_notation(), n_qubits, len(target_qubits), quantum_classical),\n",
+    "        )\n",
+    "\n",
+    "\n",
+    "check_each_possibility(equality_circuit, quantum_number, [z])"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As this is a quantum circuit, it's important to check that it works with superpositions as well. This is why we will check the uniform superposition."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def check_uniform_superposition(c, data_qubits, target_qubits, quantum_classical: bool = True):\n",
+    "    if quantum_classical:\n",
+    "        n_qubits = len(data_qubits)\n",
+    "    else:\n",
+    "        n_qubits = len(data_qubits) // 2\n",
+    "    c = cirq.Circuit(cirq.H.on_each(data_qubits) + [c])\n",
+    "    sim = cirq.Simulator()\n",
+    "    qubit_order = list(data_qubits)\n",
+    "    qubit_order += [q for q in c.all_qubits() if q not in qubit_order + target_qubits]\n",
+    "    qubit_order += target_qubits\n",
+    "    result = sim.simulate(c, qubit_order=qubit_order)\n",
+    "    result = result.dirac_notation()\n",
+    "    final = []\n",
+    "    for s in result.split('|'):\n",
+    "        if '⟩' not in s:\n",
+    "            final.append(s)\n",
+    "            continue\n",
+    "        parts = s.split('⟩')\n",
+    "        parts[0] = format_dirac(\n",
+    "            '|' + parts[0] + '⟩', n_qubits, len(target_qubits), quantum_classical\n",
+    "        )\n",
+    "        final.append(''.join(parts))\n",
+    "    print('Acting on the uniform superposition of all states we get:')\n",
+    "    print('\\t', ''.join(final))\n",
+    "\n",
+    "\n",
+    "check_uniform_superposition(equality_circuit, quantum_number, [z])"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And for the quantum-quantum case we have."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def equality_oracle_quantum_quantum(\n",
```

Comment:
Add a diagram of the output circuit. Shouldn't the cell outputs be included in the saved file?

Reply:
in all of the notebooks we clear the output